tls_codec/src/tls_vec.rs

```diff
@@ -127,38 +127,16 @@ macro_rules! impl_serialize {
         fn serialize<W: Write>(&$self, writer: &mut W) -> Result<usize, Error> {
             // Get the byte length of the content, make sure it's not too
             // large and write it out.
-            let tls_serialized_len = $self.tls_serialized_len();
-            let byte_length = tls_serialized_len - $len_len;
-
-            let max_len = <$size>::MAX as usize;
-            debug_assert!(
-                byte_length <= max_len,
-                "Vector length can't be encoded in the vector length a {} >= {}",
-                byte_length,
-                max_len
-            );
-            if byte_length > max_len {
-                return Err(Error::InvalidVectorLength);
-            }
+            let (tls_serialized_len, byte_length) = $self.get_content_lengths()?;
 
-            let mut written = (byte_length as $size).tls_serialize(writer)?;
+            let mut written =  <$size as Serialize>::tls_serialize(&(byte_length as $size), writer)?;
 
             // Now serialize the elements
             for e in $self.as_slice().iter() {
                 written += e.tls_serialize(writer)?;
             }
 
-            debug_assert_eq!(
-                written, tls_serialized_len,
-                "{} bytes should have been serialized but {} were written",
-                tls_serialized_len, written
-            );
-            if written != tls_serialized_len {
-                return Err(Error::EncodingError(format!(
```

Comment:
Why did you remove this check in `assert_written_bytes`?

Reply:
This check is present in the `assert_written_bytes` function: see [line 191 in the new file](https://github.com/RustCrypto/formats/blob/a8cd76e4ecc203f176af0acbeca490f560a67352/tls_codec/src/tls_vec.rs#L191).